backend/controller/AccountController.ts

```diff
@@ -0,0 +1,208 @@
+                'No existing User with given ID',
+                LoernwerkErrorCodes.NOT_MATCHING
+            );
+        }
+        return user;
+    }
+    /**
+     * Returns all users of the database in an array in reduced form.
+     * @returns A reduced version of all accounts
+     */
+    public static async getAllAccounts(): Promise<Partial<IUser>[]> {
+        return await DBUser.find({ select: { name: true, id: true } });
+    }
+
+    /**
+     * Deletes an account from the database. Throws an error if no account was found to delete was found
+     * @param id the id of the account
+     */
+    public static async deleteAccount(id: number): Promise<void> {
+        const user = await DBUser.findOneBy({ id: id });
+        if (user === null) {
+            throw new LoernwerkError(
+                'No existing User with given ID',
+                LoernwerkErrorCodes.NOT_MATCHING
+            );
+        }
+        user.remove();
+    }
+
+    /**
+     * Ensures that an admin exists and creates one if none exists. This function is called called by loernwerkServer.main at server startup.
+     * The Password is printed on the console
+     */
+    public static async ensureAdminAccount(): Promise<void> {
+        const user = await DBUser.findOneBy({ type: UserClass.ADMIN });
+        if (user === null) {
+            const adminUser: Partial<IUser> = {};
+            adminUser.name = 'admin';
+            adminUser.mail = 'admin@loernwerk.de';
+            const pw = crypto.randomBytes(16).toString('hex');
+            console.log(
+                'Admin account created, username: admin, mail: admin@loernwerk.de, password: ' +
+                    pw
+            );
+            adminUser.password = await this.hashPW(pw);
+            this.createNewAccount(adminUser);
+        }
+    }
+
+    /**
+     * Saves the changes to an account in the database. Only changes name, mail or password. Other changes are discarded
+     * @param data the new user
+     */
+    public static async saveAccount(data: IUser): Promise<void> {
+        const dbuser = await DBUser.findOneBy({ id: data.id });
+        if (dbuser === null) {
+            throw new LoernwerkError(
+                'No such existing User',
+                LoernwerkErrorCodes.NOT_MATCHING
+            );
+        }
+        if (
+            data.type == null &&
+            data.name === null &&
+            data.mail === null &&
+            data.password === null
+        ) {
+            return;
+        }
+        if (data.type != null) {
+            dbuser.type = data.type; //TODO: Check for Admin role if data.role === UserClass.ADMIN
+        }
+        if (data.name != null) {
+            dbuser.name = data.name;
+        }
+        if (data.mail != null) {
+            dbuser.mail = data.mail;
+        }
+        if (data.password != null) {
+            dbuser.password = await this.hashPW(data.password);
+            dbuser.save();
```

Comment:
missing await

Reply:
we might should also save if something else is changed except the password btw